railib/rest.py

```diff
@@ -210,6 +214,28 @@ def _authenticate(ctx: Context, req: Request) -> Request:
         return req
     raise Exception("unknown credential type")
 
+# Issues an HTTP request and retries if failed due to URLError.
+def _urlopen_with_retry(req: Request, retries: int = 0):
+    if retries < 0:
+        raise ValueError("Retries must be a non-negative integer")
+
+    attempts = retries + 1
+
+    for attempt in range(attempts):
+        try:
+            return urlopen(req)
+        except URLError as e:
+            if isinstance(e.reason, socket.timeout):
+                logger.warning(f"Timeout occurred (attempt {attempt + 1}/{attempts}): {req.full_url}")
+            else:
+                logger.warning(f"URLError occurred {e.reason} (attempt {attempt + 1}/{attempts}): {req.full_url}")
+
+        if retries > 0:
+            time.sleep(attempt + 1)
```

Comment:
I don't think there's any need to delay at all here.. a brief review of urllib3 in Python which implements retries doesn't seem to have any.  Generally speaking I think the most common case of timing out/discos already puts the spacing into the seconds, so the value of a delay here is not very clear I think